src/NetscriptFunctions/CodingContract.ts

```diff
@@ -59,11 +69,31 @@ export function NetscriptCodingContract(): InternalAPI<ICodingContract> {
       const contract = getCodingContract(ctx, hostname, filename);
       return contract.getType();
     },
-    getData: (ctx) => (_filename, _hostname?) => {
+    getData:
+      (ctx) =>
+      (_filename: unknown, _hostname?: unknown, _type?: unknown): any => {
+        const filename = helpers.string(ctx, "filename", _filename);
+        const hostname = _hostname ? helpers.string(ctx, "hostname", _hostname) : ctx.workerScript.hostname;
+        const contract = getCodingContract(ctx, hostname, filename);
+        if (_type) {
+          const type = helpers.string(ctx, "type", _type);
+          if (contract.type !== type)
+            throw helpers.errorMessage(
+              ctx,
+              `The given type '${type}' is not the same as the one of contract '${contract.fn}' of type '${contract.type}'.`,
+            );
+        }
+        return structuredClone(contract.getData());
+      },
+    getContract: (ctx) => (_filename: unknown, _hostname?: unknown) => {
       const filename = helpers.string(ctx, "filename", _filename);
       const hostname = _hostname ? helpers.string(ctx, "hostname", _hostname) : ctx.workerScript.hostname;
       const contract = getCodingContract(ctx, hostname, filename);
-      return structuredClone(contract.getData());
+      // asserting type here is required, since it is not feasible to properly type getData
+      return {
+        type: contract.type,
```

Comment:
What about the other attributes? I would expect description, tires remaining, etc. to be in here.